lib/Stomp/Server.pm6

```diff
@@ -1,26 +1,116 @@
 use Stomp::Parser;
 use Stomp::MessageStream;
+need Stomp::Message;
 
-class Stomp::Server does Stomp::MessageStream[Stomp::Parser::ClientCommands] {
+class Stomp::Server {
     has Str $.host is required;
     has Int $.port is required;
 
-    method listen() {
-        supply {
-            whenever self.socket-provider.listen($!host, $!port) -> $conn {
-                self!process-messages($conn).tap:
+    enum AckMode ( Auto => 'auto', Client => 'client', Individual => 'client-individual');
+
+    class Subscription {
+        has Str $.id            is required;
+        has Str $.destination   is required;
+        has AckMode $.ack = Auto;
+
+        method ACCEPTS(Stomp::Message $mess) {
+            $!destination eq $mess.headers<destination>;
+        }
+    }
+
+    class Connection does Stomp::MessageStream[Stomp::Parser::ClientCommands] {
+        has $.conn;
+        has Supply $!messages;
+
+        has  Subscription @.subscriptions;
+        has  Lock $!subscription-lock;
+
+        has Supplier $!sent-message-supplier;
+        has Promise $.connected;
+
+        submethod TWEAK {
+            $!messages = self!process-messages($!conn.Supply).share;
+
+            $!subscription-lock = Lock.new;
+
+            $!connected = Promise.new;
+
+            my &quit = {
+                when X::Stomp::MalformedMessage {
+                    await $!conn.print: Stomp::Message.new:
+                                command => 'ERROR',
+                                body => .message;
+                }
+            };
+
+            my $connect-tap = $!messages.grep({ $_.command ~~ 'CONNECT'|'STOMP' }).tap:
                     {
-                        await $conn.print: Stomp::Message.new:
+                        await $!conn.print: Stomp::Message.new:
                             command => 'CONNECTED',
                             headers => ( accept-version => '1.2' );
```

Comment:
The 1.2 spec indicates that which of 'CONNECT' or 'STOMP' is used is a heuristic to which version it is.  'STOMP' indicates 1.2 

Also this should be 'version' not 'accept-version'